app/View/Components/Community/UserCard.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\View\Components\Community;
+
+use App\Community\Enums\Rank;
+use App\Community\Enums\RankType;
+use App\Site\Enums\Permissions;
+use App\Site\Models\User as UserModel;
+use App\Support\Cache\CacheKey;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\View\Component;
+
+class UserCard extends Component
+{
+    private string|array $user;
+
+    public function __construct(string|array $user)
+    {
+        $this->user = $user;
+    }
+
+    public function render(): ?View
+    {
+        $username = $this->getUsername($this->user);
+
+        if (empty($username)) {
+            return null;
+        }
+
+        /* @var array $rawUserData */
+        $rawUserData = [];
+        if (is_array($this->user)) {
+            $rawUserData = $this->user;
+        } else {
+            $rawUserData = $this->getUserData($username);
+        }
+
+        if (!$rawUserData) {
+            return null;
+        }
+
+        $cardViewValues = $this->buildAllCardViewValues($username, $rawUserData);
+
+        return view('components.community.user-card', $cardViewValues);
+    }
+
+    private function getUsername(string|array $user): ?string
+    {
+        return is_string($user) ? $user : ($user['User'] ?? null);
+    }
+
+    private function getUserData(string $username): ?array
+    {
+        return Cache::store('array')->remember(
+            CacheKey::buildUserCardDataCacheKey($username . "1"),
+            Carbon::now()->addMonths(3),
+            function () use ($username): ?array {
+                $foundUser = UserModel::firstWhere('User', $username);
+
+                return $foundUser ? $foundUser->toArray() : null;
+            }
+        );
+    }
+
+    private function buildAllCardViewValues(string $username, array $rawUserData): array
+    {
+        $cardBioData = $this->buildCardBioData($rawUserData);
+        $cardRankData = $this->buildCardRankData($username, $rawUserData['RAPoints'], $rawUserData['RASoftcorePoints'], $rawUserData['Untracked'] ? true : false);
+        $cardRoleData = $this->buildCardRoleData($username, (int) $rawUserData['Permissions']);
+
+        return array_merge($cardBioData, $cardRankData, $cardRoleData);
+    }
+
+    private function buildCardBioData(array $rawUserData): array
+    {
+        $username = $rawUserData['User'] ?? "";
+        $motto = $rawUserData['Motto'] ?? null;
+        $avatarUrl = $rawUserData['avatarUrl'] ?? null;
+        $hardcorePoints = $rawUserData['RAPoints'] ?? 0;
+        $softcorePoints = $rawUserData['RASoftcorePoints'] ?? 0;
+        $retroPoints = $rawUserData['TrueRAPoints'] ?? 0;
+        $isUntracked = $rawUserData['Untracked'] ? true : false;
+        $permissions = $rawUserData['Permissions'] ?? Permissions::Unregistered;
+        $memberSince = $rawUserData['Created'] ?? Carbon::now();
```

Comment:
I'm assuming Created isn't visible, as all my users have been members since today.
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/46f3ef70-ca3b-46b2-b1f7-6980a3dbd499)

Reply:
Thanks, fixed in https://github.com/RetroAchievements/RAWeb/pull/1674/commits/87b59e67ccb06ac73265d9fd3adbc75ac87d2ed4.